### PR TITLE
feat: adding posthog libs for node + js

### PR DESCRIPTION
**Pull request by abhi-slash-git**
## What does this PR do?

- Adds posthog node + js libs for event tracking. 

## Mandatory Tasks (DO NOT REMOVE)

- [ ] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

**Comment by abhi-slash-git** [comment]
@sean-brydon PR created, would like your help with the frontend setup. Thank you

**Comment by sean-brydon** [comment]
> @sean-brydon PR created, would like your help with the frontend setup. Thank you

Posthog is now initializing on the frontend and saving page views. 

We should think when we want to call the identify function on the user - in shell was re-rendering too much. Maybe we do this upon login?

**Comment by abhi-slash-git** [comment]
@sean-brydon Yep you are right, we can do this on login. We also have to remember to do call reset() on logout as well.

File: packages/features/ee/event-tracking/lib/posthog/userPostHog.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      metadata: data?.metadata,
+      completed_onboarding: data?.completedOnboarding,
+      sum_of_bookings: data?.sumOfBookings,
+      sum_of_calendars: data?.sumOfCalendars,
+      sum_of_teams: data?.sumOfTeams,
+      has_orgs_plan: !!data?.organizationId,
+      organization: data?.organization?.slug,
+      sum_of_event_types: data?.sumOfEventTypes,
+      sum_of_team_event_types: data?.sumOfTeamEventTypes,
+      is_premium: data?.isPremium,
```
Comment (abhi-slash-git) [inline packages/features/ee/event-tracking/lib/posthog/userPostHog.tsx:59]: @sean-brydon check which properties to remove

File: packages/features/auth/lib/next-auth-options.ts [hunk 2]
```diff
@@ -947,24 +948,32 @@ export const getOptions = ({
...
+                customerId: user.id.toString(),
+                customerName: user.name,
+                customerEmail: user.email,
+                customerAvatar: user.image,
+              })
+            );
+          }
         }
+
+        postHogClient().capture(user.id.toString(), "Sign Up", {
```
Comment (abhi-slash-git) [inline packages/features/auth/lib/next-auth-options.ts:972]: @sean-brydon change this to the correct event name

**Comment by abhi-slash-git** [comment]
@sean-brydon bump for visibilty

**Review by sean-brydon — COMMENTED** [review]
LGTM - will see if someone else can have some eyes as i did a bit of work on this also

File: packages/features/ee/event-tracking/lib/posthog/provider.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+import posthog from "posthog-js";
+import { PostHogProvider } from "posthog-js/react";
+import { useEffect } from "react";
+
+function Provider({ children }: { children: React.ReactNode }) {
+  useEffect(() => {
+    if (!process.env.NEXT_PUBLIC_POSTHOG_KEY) return;
+    posthog.init(process.env.NEXT_PUBLIC_POSTHOG_KEY, {
+      api_host: process.env.NEXT_PUBLIC_POSTHOG_HOST || "https://us.i.posthog.com",
```
Comment (baileypumfleet) [inline packages/features/ee/event-tracking/lib/posthog/provider.tsx:11]: Even though we use the EU PostHog region, we can keep the US one as the fallback

**Review by pumfleet — APPROVED** [review]
Code looks good to me

Thank you for your help on this :)

**Comment by baileypumfleet** [comment]
@abhi-slash-git could you sign the CLA real quick? You should see it as the second comment on this PR

**Comment by baileypumfleet** [comment]
Screenshot 2024-11-04 at 11 25 39
Just noticed the text is supposed to be grey - figma is above, website is below
Screenshot 2024-11-04 at 11 26 00

**Comment by neuhaus** [comment]
i guess changing the line to
`      if (searchParams?.toString()) {`
will fix the issue?

**Comment by neuhaus** [comment]
@abhi-slash-git i don't understand why you didn't have this issue during building

**Comment by abhi-slash-git** [comment]
@sean-brydon might know, I believe he added this change

**Comment by zomars** [comment]
@neuhaus @abhi-slash-git  @sean-brydon this is now causing issues locally. I'm getting the same errors when building locally.

image

**Pull request merged**